HopsanGenerator/src/generators/HopsanExeGenerator.cpp

```diff
@@ -0,0 +1,375 @@
+    // Add HopsanCore (and necessary dependency) include paths
+    Q_FOREACH(const QString &incPath, getHopsanCoreIncludePaths())
+    {
+       compileCppBatchStream << QString(" -I\"%1\"").arg(incPath);
+    }
+    compileCppBatchFile.close();
+
+    cppCompileOK = callProcess("/bin/sh", QStringList() << "compileCpp.sh", buildPath);
+#endif
+    if (!cppCompileOK) {
+        printErrorMessage("Failed to compile exported C++ Hopsan code for executable model.");
+        return false;
+    }
+
+    QStringList objectFiles;
+    objectFiles << "exe_main.o" << "exe_utilities.o";
+    for(const QString& extraSrc : mExtraSourceFiles) {
+        QFileInfo extraObjFile(extraSrc);
+        objectFiles << extraObjFile.baseName()+".o";
+    }
+    for(const QString &srcFile : srcFiles) {
+        QFileInfo fi(srcFile);
+        objectFiles << fi.baseName()+".o";
+    }
+
+    if(!assertFilesExist(buildPath, objectFiles)) {
+        return false;
+    }
+
+    printMessage("------------------------------------------------------------------------");
+    printMessage("Linking");
+    printMessage("------------------------------------------------------------------------");
+
+    bool linkingOK=false;
+
+    QString outputExecutableFile;
+#ifdef _WIN32
+    outputExecutableFile = QString("%1/%2.exe").arg(buildPath).arg(modelName);
+#elif __linux__
+    outputExecutableFile = QString("%1/%2").arg(buildPath).arg(modelName);
+#endif
+    //! @todo Add OSX support
+    // Create output directory before linking
+    QDir systemRootDir("");
+    systemRootDir.mkpath(QFileInfo(outputExecutableFile).absolutePath());
+
+#ifdef _WIN32
+    QFile linkBatchFile;
+    linkBatchFile.setFileName(buildPath + "/link.bat");
+    if(!linkBatchFile.open(QIODevice::WriteOnly | QIODevice::Text))
+    {
+        printErrorMessage("Failed to open link.bat for writing.");
+        return false;
+    }
+    //Write the compilation script file
+    QTextStream linkBatchStream(&linkBatchFile);
+    linkBatchStream << "@echo off\n";
+    linkBatchStream << "PATH=" << mCompilerSelection.path << ";%PATH%\n";
+    linkBatchStream << "@echo on\n";
+    linkBatchStream << "g++ -w -static -static-libgcc -fPIC -Wl,--rpath,'$ORIGIN/.' ";
+    Q_FOREACH(const QString &objFile, objectFiles)
+    {
+        linkBatchStream << " " << objFile;
+    }
+    linkBatchStream << " -o "+outputExecutableFile+"\n";
+    linkBatchFile.close();
+
+    linkingOK = callProcess("cmd.exe", QStringList() << "/c" << "cd /d " + buildPath + " & link.bat");
+
+#else
+    QFile linkBatchFile;
+    linkBatchFile.setFileName(buildPath + "/link.sh");
+    if(!linkBatchFile.open(QIODevice::WriteOnly | QIODevice::Text))
+    {
+        printErrorMessage("Failed to open link.sh for writing.");
+        return false;
+    }
+    //Write the compilation script file
+    QTextStream linkBatchStream(&linkBatchFile);
+    linkBatchStream << mCompilerSelection.path+"g++ -pthread -fPIC -w -static-libgcc -Wl,--rpath,'$ORIGIN/.'";
```

Comment:
If everything is linked statically then setting rpath here should not be needed, but I dont know if it should be possible to load external "dll libraries" or not

Reply:
That is not the intention. I will try with removing rpath. The purpose of this export is to obtain only one single exe file.